unittest/serializer/SerializerUnittest.cpp

```diff
@@ -128,20 +135,20 @@ TEST_F(SerializerUnittest, Write)
         fs.seekg(0, fs.end);
         int filelength = fs.tellg();
         fs.seekg(0, fs.beg);
-        ASSERT_EQ(sizeof(double)*jsize*ksize, filelength);
+        ASSERT_EQ(sizeof(float)*jsize*ksize, filelength);
 
         std::vector<char> rawdata(filelength);
         fs.read(rawdata.data(), filelength);
         fs.close();
 
         // Check data
-        const double* pNewData = reinterpret_cast<const double*>(rawdata.data());
+        const float* pNewData = reinterpret_cast<const float*>(rawdata.data());
```

Comment:
Ah I think I see. I think we should have different tests for single and double precision. I would split the test function and then template it in single and double precision.